### ISSUE TITLE
Guest Checkout into core

### ISSUE THREAD

**Issue by barco57** [zencart/zencart#2689]
I was going title this issue "What I want, What I really really want"
So since there seems to be a fair bit of issues for feature requests I thought I would post my most wanted addition to the core. 

What I really want in the core is the guest checkout portion of @lat9 One Page Checkout module. What I particularly like about her guest checkout is that it doesn't create a new customer for each order, it uses a "phantom" customer for all guest checkout orders, keeps thing much cleaner. 

Recently I ran into an issue that related to wanting guest checkout on a client site. They wanted an upgrade and really wanted guest checkout. I opted for the ZCA Bootstrap template (my favorite) and OPC to get the guest checkout. However I ran into a problem, to use guest checkout I have to turn on one page check out, which can be a bit particular about how the page is put together, and ZCA Bootstrap breaks it. To quote my correspondence "the bootstrap template has ‘tossed’ many of the HTML id’s that are used as selectors by OPC’s jQuery processing.". So I have ended up turning off OPC on that site for now until I decide how best to move forward with getting guest checkout... preferably without using COWAA or resorting to using Responsive Classic (which I hate). Lat9 having pity on me and creating the correct ZCA Bootstrap overrides for her module would be nice but opens up a can of worms where everybody who makes templates will want her to do their overrides, so that's a non-starter. rbarbour may do it, he used to ship OPC overrides but removed them, and has said he will add them back as a separate repo at some point but as most have noticed he likes to take long breaks (he is a busy guy) and he is not getting rich off the template, so I can't beat him up over it. So the only way I can see to solve it, for myself, is to see if I can figure out how to get guest checkout into core and then everybody has to make their stuff work with it....yeah, that's going to be a popular sentence... ok, me being whiny about what I want is hopefully over, been a long summer already with three teenagers and thought I might get some gimmee, gimmee, gimmee posting in so I can join their club...LOL

Ok, discuss

**Comment by wiztechinc** [comment]
I think something of this ilk should be in core though I get fewer requests for it now than I did 5 years ago. I don't like what I've been using as it is confusing to customers and buyers alike because of the way it goes into the database. I haven't tried the one page checkout mentioned cause the earlier one by numinix was just disastrous. The phantom customer is where we need to go for sure. I had COWOA in my template but have removed it so I no longer have core changes. When given the choice I think maybe half turned it on and half didn't.

**Comment by lankeeyankee** [comment]
Not sure about a generic customer. I tried OPC and went back to FEC. There are many instances of needing the actual customer's details. Especially with followup phone orders and using EMP.

**Comment by lankeeyankee** [comment]
But I very much agree to the concept of including guest checkout into the core, I'd call it essential based on our experience of more than 50% of all orders being guest checkout. And offering the guest the option of creating an account on the checkout success page, that is important, too.

**Comment by mc12345678** [comment]
Regarding hooks into the code, now is the time to recommend them for ZC 1.5.7 to reduce/eliminate core code changes whether it is for the associated guest checkout or for other useful plugins.

**Comment by barco57** [comment]
> Not sure about a generic customer. I tried OPC and went back to FEC. There are many instances of needing the actual customer's details. Especially with followup phone orders and using EMP.

all of those contact details are on the order. If I am going to contact the customer about the order I am most likely going to have their order open rather than their customer page

**Comment by wiztechinc** [comment]
It's amazing to me how many folks never even go to admin but only use the order when it comes in email. Literally printing the email and handing it off to someone else to fill.

**Comment by lankeeyankee** [comment]
> 
> all of those contact details are on the order. If I am going to contact the customer about the order I am most likely going to have their order open rather than their customer page

True, but what about when they come back and order again over the phone? Do you have to re-enter everything again? I have not used OPC deeply enough to know the answer.

**Comment by barco57** [comment]
yes, you would have to enter it again, because they used guest checkout you can not login to the account on the front end as it doesn't really exist... and even with the other COWOA module you can not login to the process an order from the front end... that is sort of the point of guest checkout.

**Comment by lankeeyankee** [comment]
I can totally log in with EMP? We do it every day. And when it's a phone customer's first order we always do it as a guest. And when they come back, we just go to the login, enter their email and our password, and it's on to the checkout page. We use FEC with COWOA enabled.

**Comment by drbyte** [comment]
> @barco57 wrote:
>  I opted for the ZCA Bootstrap template (my favorite) and OPC to get the guest checkout. However I ran into a problem, to use guest checkout I have to turn on one page check out, which can be a bit particular about how the page is put together, and ZCA Bootstrap breaks it.

Do OPC and ZCA Bootstrap play together nicely yet?

**Comment by barco57** [comment]
No, there has been no movement on that front.

**Comment by scottcwilson** [comment]
This feature was built in to 2.0.  Perhaps we could backport it to 1.5.8 or pull in @lat9's one page checkout mod (which includes Guest Checkout) into the core.  Guest Checkout is a frequently requested feature.

**Comment by torvista** [comment]
> pull in @lat9's one page checkout mod (which includes Guest Checkout) into the core

In principle yes, but having been subscribed to the thread for a while, it seems there are constant issues cropping up (due to user customisations), despite the maturity of the plugin.
So, for me, it should be pulled in, but when lat9 says so!
